### PR TITLE
Spawns a thread to run PS4 app

### LINKED ISSUE #295: Find a solution for specifying stack address when creating a thread on Windows
The syscall for creating a thread on the FreeBSD and PS4 can specify stack address. The problem is Windows does not allow this. The closed thing I can find is creating a thread in suspended state then use `SetThreadContext` to set `RSP` directly. The problem is it seems like this will break SEH.

### PR DESCRIPTION
**Pull request by ultimaweapon**
Resolves #295.

### CODE REVIEW

**Comment by ultimaweapon** [comment]
@VocalFan can you test this to see if everything is still working as usual?

**Comment by VocalFan** [comment]
Exact same issue and behavior as #297 https://github.com/obhq/obliteration/pull/297#issuecomment-1670546824

`24DF86B99B6: 48 89 8C C4 E0 00 00 00    mov    qword ptr [rsp + 8*rax + 0xe0], rcx`
`Stop reason: Exception 0xc0000005 encountered at address 0x24df86b99b6: Unknown access violation (code 3) at location 0x00000000`

**Comment by ultimaweapon** [comment]
Strange.

**Comment by VocalFan** [comment]
And I know I'm not using the same code, as besides from using Github Actions to change branches to this PR, I also tested both PR's dev builds from the Actions... Still, it works until it reaches Syscall 608. As all the other ones that end early at Syscall 599's missing Relocation type, *or We Are Doomed's SELF error*, all properly give Int 0x44 instead of "The Kernel Crashed"

**Comment by VocalFan** [comment]
So, after extended spamming... It very rarely works... But it does.

image

**Comment by ultimaweapon** [comment]
It seems like the access violation came from Rust panic. Looks like Rust panic overrun the stack somehow. With system provided stack this will trigger stack buffer overrun that you have experienced before but for our own stack it will cause access violation because we don't have a guard page.

**Comment by VocalFan** [comment]
So... It is working... Kinda?

**Comment by VocalFan** [comment]
I recommend focusing on #297 as it is a smaller change and... Well... Looking at the ReadME, doesn't require python haha.

**Comment by ultimaweapon** [comment]
Python is required by LLVM, not us. This one is better than #297 because we don't need to switch stack when crossing the boundary.

**Comment by VocalFan** [comment]
Ah, well, I thought Python was required by this PR mainly because well... This one has the ReadME change, while 297 doesn't haha. But if so, alrighty!

**Comment by ultimaweapon** [comment]
Just reported this issue to Rust: https://github.com/rust-lang/rust/issues/114680

**Comment by VocalFan** [comment]
> Just reported this issue to Rust: [rust-lang/rust#114680](https://github.com/rust-lang/rust/issues/114680)

Gotta love how the language itself is open-source, makes it easy to fix bugs! ...Tell me if you need anything else, I'll be online!

**Comment by VocalFan** [comment]
@ultimaweapon Seems the error might be on our side to handle, look at the issue you made for rust.

**Comment by ultimaweapon** [comment]
I just replied them.

**Comment by ultimaweapon** [comment]
@VocalFan can you try again to see if no more access violation?

**Comment by VocalFan** [comment]
> @VocalFan can you try again to see if no more access violation?

Built from Source Code, seems to have exact same behavior.

<details>
  <summary>
    Persona 5 Royal log:
  </summary>
  <code>++++++++++++++++++ I [00:00:00:00:004]:0x00000000000059e8:kernel\src/main.rs:84
Starting Obliteration Kernel.
System directory    : G:\obSTORAGE\dev_sys\
Game directory      : G:\obSTORAGE\dev_hdd0\CUSA12380
Debug dump directory: G:\obSTORAGE\log_dev
++++++++++++++++++ I [00:00:00:00:005]:0x00000000000059e8:kernel\src/main.rs:97
Initializing arc4random.
++++++++++++++++++ I [00:00:00:00:005]:0x00000000000059e8:kernel\src/main.rs:101
Initializing LLVM.
++++++++++++++++++ I [00:00:00:00:006]:0x00000000000059e8:kernel\src/main.rs:107
Initializing file system.
++++++++++++++++++ I [00:00:00:00:010]:0x00000000000059e8:kernel\src/main.rs:118
Initializing memory manager.
++++++++++++++++++ I [00:00:00:00:011]:0x00000000000059e8:kernel\src/main.rs:121
Page size is             : 4096
Allocation granularity is: 65536
++++++++++++++++++ I [00:00:00:00:012]:0x00000000000059e8:kernel\src/main.rs:134
Initializing virtual process.
++++++++++++++++++ I [00:00:00:00:012]:0x00000000000059e8:kernel\src/main.rs:139
Initializing runtime linker.
++++++++++++++++++ I [00:00:00:00:020]:0x00000000000059e8:kernel\src/main.rs:150
Application   : /mnt/sandbox/CUSA12380_000/app0/eboot.bin
Image format  : SELF
Image type    : ET_SCE_DYNEXEC
Program 0     : 0x0000000000004000:0x0000000000595a18 => 0x0000000000000000:0x0000000000591a18 PT_LOAD
Program 1     : 0x0000000000598000:0x00000000005d8438 => 0x0000000000594000:0x00000000005d8000 PT_SCE_RELRO
Program 2     : 0x00000000005dc000:0x00000000007dbb58 => 0x00000000005d8000:0x0000000000a41120 PT_LOAD
Program 3     : 0x00000000005dc000:0x00000000005dc050 => 0x00000000005d8000:0x00000000005d8050 PT_SCE_PROCPARAM
Program 4     : 0x00000000008ac470:0x00000000008ad2b0 => 0x0000000000000000:0x0000000000000e40 PT_DYNAMIC
Program 5     : 0x0000000000004000:0x0000000000004015 => 0x0000000000000000:0x0000000000000015 PT_INTERP
Program 6     : 0x0000000000000000:0x0000000000000000 => 0x0000000000000000:0x0000000000000000 PT_TLS
Program 7     : 0x0000000000571f54:0x0000000000595a18 => 0x000000000056df54:0x0000000000591a18 PT_GNU_EH_FRAME
Program 8     : 0x00000000007dbb60:0x00000000008ad2b0 => 0x0000000000000000:0x0000000000000000 PT_SCE_DYNLIBDATA
Program 9     : 0x00000000008ad2b0:0x00000000008ad308 => 0x0000000000000000:0x0000000000000000 PT_SCE_COMMENT
Program 10    : 0x00000000008ad308:0x00000000008adb81 => 0x0000000000000000:0x0000000000000000 PT_SCE_VERSION
Needed        : libSceErrorDialog.prx
Needed        : libSceSysmodule.prx
Needed        : libSceUserService.prx
Needed        : libSceSystemService.prx
Needed        : libSceFiber.prx
Needed        : libSceFios2.prx
Needed        : libScePad.prx
Needed        : libSceVideoOut.prx
Needed        : libSceGnmDriver.prx
Needed        : libkernel.prx
Needed        : libSceMouse.prx
Needed        : libSceIme.prx
Needed        : libSceRtc.prx
Needed        : libSceFont-module.prx
Needed        : libSceFontFt-module.prx
Needed        : libSceAudioOut.prx
Needed        : libSceSaveData.prx
Needed        : libSceSaveDataDialog.prx
Needed        : libSceCommonDialog.prx
Needed        : libSceScreenShot.prx
Needed        : libSceGameLiveStreaming.prx
Needed        : libSceVideoRecording.prx
Needed        : libSceVideodec.prx
Needed        : libSceNet.prx
Needed        : libSceNetCtl.prx
Needed        : libSceNpCommon.prx
Needed        : libSceNpManager.prx
Needed        : libSceNpScoreRanking.prx
Needed        : libSceNpTrophy.prx
Needed        : libSceNpTus.prx
Needed        : libSceNpMatching2.prx
Needed        : libSceSsl.prx
Needed        : libSceHttp.prx
Needed        : libSceNpWebApi.prx
Needed        : libSceAppContent.prx
Needed        : libScePlayGo.prx
Needed        : libScePlayGoDialog.prx
Needed        : libSceImeBackend.prx
Needed        : libSceSharePlay.prx
Needed        : libSceJson2.prx
Needed        : libSceSigninDialog.prx
Needed        : libSceMsgDialog.prx
Needed        : libSceHmd.prx
Needed        : libSceCamera.prx
Needed        : libSceVrTracker.prx
Needed        : libSceHmdSetupDialog.prx
Needed        : libSceAjm.prx
Needed        : libSceNpCommerce.prx
Needed        : libc.prx
Module flags  : MAIN_PROG | TEXT_REL | UNK2
TLS index     : 1
Memory address: 0x000001dac5e30000:0x000001dac6e74000
Initialization: 0x000001dac6230020
Entry address : 0x000001dac6450730
Finalization  : 0x000001dac663ab00
TLS init      : 0x000001dac6230000:0x000001dac6230000
TLS size      : 0
EH header     : 0x000001dac679df54:0x000001dac67c1a18
EH frame      : 0x000001dac671d3d0:0x000001dac679df54
Process param : 0x000001dac6808000:0x000001dac6808050
Program 0 is mapped to 0x000001dac6230000:0x000001dac67c4000 with EXECUTE | READ.
Program 1 is mapped to 0x000001dac67c4000:0x000001dac6808000 with READ.
Program 2 is mapped to 0x000001dac6808000:0x000001dac6c74000 with WRITE | READ.
++++++++++++++++++ I [00:00:00:00:025]:0x00000000000059e8:kernel\src/main.rs:158
Loading /system/common/lib/libkernel.sprx.
++++++++++++++++++ I [00:00:00:00:027]:0x00000000000059e8:kernel\src/main.rs:169
Image format  : ELF
Image type    : ET_SCE_DYNAMIC
Program 0     : 0x0000000000004000:0x0000000000054318 => 0x0000000000000000:0x0000000000050318 PT_LOAD
Program 1     : 0x0000000000058000:0x0000000000059010 => 0x0000000000054000:0x0000000000055010 PT_SCE_RELRO
Program 2     : 0x000000000005c000:0x000000000005c330 => 0x0000000000058000:0x000000000008f590 PT_LOAD
Program 3     : 0x000000000005c000:0x000000000005c020 => 0x0000000000058000:0x0000000000058020 PT_SCE_MODULE_PARAM
Program 4     : 0x000000000006d488:0x000000000006d9a8 => 0x0000000000000000:0x0000000000000520 PT_DYNAMIC
Program 5     : 0x0000000000000000:0x0000000000000000 => 0x0000000000000000:0x0000000000000000 PT_TLS
Program 6     : 0x0000000000050e9c:0x0000000000054318 => 0x000000000004ce9c:0x0000000000050318 PT_GNU_EH_FRAME
Program 7     : 0x000000000005c330:0x000000000006d9a8 => 0x0000000000000000:0x0000000000000000 PT_SCE_DYNLIBDATA
Program 8     : 0x0000000000000000:0x0000000000000000 => 0x0000000000000000:0x0000000000000000 PT_SCE_VERSION
Needed        : libSceLibcInternal.prx
Needed        : libSceGnmDriver.prx
Module flags  : UNK1
TLS index     : 0
Memory address: 0x000001dac6e80000:0x000001dac7110000
Entry address : 0x000001dac6ea87e8
Finalization  : 0x000001dac6eb67a0
TLS init      : 0x000001dac6e80000:0x000001dac6e80000
TLS size      : 0
EH header     : 0x000001dac6ecce9c:0x000001dac6ed0318
EH frame      : 0x000001dac6ebc270:0x000001dac6ecce9c
Program 0 is mapped to 0x000001dac6e80000:0x000001dac6ed4000 with EXECUTE | READ.
Program 1 is mapped to 0x000001dac6ed4000:0x000001dac6ed8000 with READ.
Program 2 is mapped to 0x000001dac6ed8000:0x000001dac6f10000 with WRITE | READ.
++++++++++++++++++ I [00:00:00:00:029]:0x00000000000059e8:kernel\src/main.rs:180
Loading /system/common/lib/libSceLibcInternal.sprx.
++++++++++++++++++ I [00:00:00:00:031]:0x00000000000059e8:kernel\src/main.rs:191
Image format  : ELF
Image type    : ET_SCE_DYNAMIC
Program 0     : 0x0000000000004000:0x0000000000129c8c => 0x0000000000000000:0x0000000000125c8c PT_LOAD
Program 1     : 0x000000000012c000:0x0000000000130510 => 0x0000000000128000:0x000000000012c510 PT_SCE_RELRO
Program 2     : 0x0000000000134000:0x000000000013dbe8 => 0x0000000000130000:0x0000000000149110 PT_LOAD
Program 3     : 0x0000000000134000:0x000000000013a4d8 => 0x0000000000130000:0x00000000001364d8 PT_SCE_MODULE_PARAM
Program 4     : 0x000000000016cc78:0x000000000016cf18 => 0x0000000000000000:0x00000000000002a0 PT_DYNAMIC
Program 5     : 0x000000000013a4e0:0x000000000013a660 => 0x00000000001364e0:0x0000000000136a40 PT_TLS
Program 6     : 0x0000000000122a88:0x0000000000129c8c => 0x000000000011ea88:0x0000000000125c8c PT_GNU_EH_FRAME
Program 7     : 0x000000000013dc00:0x000000000016cf18 => 0x0000000000000000:0x0000000000000000 PT_SCE_DYNLIBDATA
Program 8     : 0x0000000000000000:0x0000000000000000 => 0x0000000000000000:0x0000000000000000 PT_SCE_VERSION
Needed        : libkernel.prx
Needed        : libSceLibcInternal.prx
Module flags  : UNK1
TLS index     : 2
Memory address: 0x000001dac7110000:0x000001dac745c000
Finalization  : 0x000001dac71e53b0
TLS init      : 0x000001dac72464e0:0x000001dac7246660
TLS size      : 1376
EH header     : 0x000001dac722ea88:0x000001dac7235c8c
EH frame      : 0x000001dac7209664:0x000001dac722ea88
Program 0 is mapped to 0x000001dac7110000:0x000001dac7238000 with EXECUTE | READ.
Program 1 is mapped to 0x000001dac7238000:0x000001dac7240000 with READ.
Program 2 is mapped to 0x000001dac7240000:0x000001dac725c000 with WRITE | READ.
++++++++++++++++++ I [00:00:00:00:033]:0x00000000000059e8:kernel\src/main.rs:194
Initializing system call routines.
++++++++++++++++++ I [00:00:00:00:034]:0x00000000000059e8:kernel\src/main.rs:200
Initializing execution engine.
++++++++++++++++++ I [00:00:00:00:000]:0x0000000000000258:kernel\src/main.rs:84
Starting Obliteration Kernel.
System directory    : G:\obSTORAGE\dev_sys\
Game directory      : G:\obSTORAGE\dev_hdd0\CUSA12380
Debug dump directory: G:\obSTORAGE\log_dev
++++++++++++++++++ I [00:00:00:00:001]:0x0000000000000258:kernel\src/main.rs:97
Initializing arc4random.
++++++++++++++++++ I [00:00:00:00:001]:0x0000000000000258:kernel\src/main.rs:101
Initializing LLVM.
++++++++++++++++++ I [00:00:00:00:002]:0x0000000000000258:kernel\src/main.rs:107
Initializing file system.
++++++++++++++++++ I [00:00:00:00:006]:0x0000000000000258:kernel\src/main.rs:118
Initializing memory manager.
++++++++++++++++++ I [00:00:00:00:006]:0x0000000000000258:kernel\src/main.rs:121
Page size is             : 4096
Allocation granularity is: 65536
++++++++++++++++++ I [00:00:00:00:007]:0x0000000000000258:kernel\src/main.rs:134
Initializing virtual process.
++++++++++++++++++ I [00:00:00:00:007]:0x0000000000000258:kernel\src/main.rs:139
Initializing runtime linker.
++++++++++++++++++ I [00:00:00:00:016]:0x0000000000000258:kernel\src/main.rs:150
Application   : /mnt/sandbox/CUSA12380_000/app0/eboot.bin
Image format  : SELF
Image type    : ET_SCE_DYNEXEC
Program 0     : 0x0000000000004000:0x0000000000595a18 => 0x0000000000000000:0x0000000000591a18 PT_LOAD
Program 1     : 0x0000000000598000:0x00000000005d8438 => 0x0000000000594000:0x00000000005d8000 PT_SCE_RELRO
Program 2     : 0x00000000005dc000:0x00000000007dbb58 => 0x00000000005d8000:0x0000000000a41120 PT_LOAD
Program 3     : 0x00000000005dc000:0x00000000005dc050 => 0x00000000005d8000:0x00000000005d8050 PT_SCE_PROCPARAM
Program 4     : 0x00000000008ac470:0x00000000008ad2b0 => 0x0000000000000000:0x0000000000000e40 PT_DYNAMIC
Program 5     : 0x0000000000004000:0x0000000000004015 => 0x0000000000000000:0x0000000000000015 PT_INTERP
Program 6     : 0x0000000000000000:0x0000000000000000 => 0x0000000000000000:0x0000000000000000 PT_TLS
Program 7     : 0x0000000000571f54:0x0000000000595a18 => 0x000000000056df54:0x0000000000591a18 PT_GNU_EH_FRAME
Program 8     : 0x00000000007dbb60:0x00000000008ad2b0 => 0x0000000000000000:0x0000000000000000 PT_SCE_DYNLIBDATA
Program 9     : 0x00000000008ad2b0:0x00000000008ad308 => 0x0000000000000000:0x0000000000000000 PT_SCE_COMMENT
Program 10    : 0x00000000008ad308:0x00000000008adb81 => 0x0000000000000000:0x0000000000000000 PT_SCE_VERSION
Needed        : libSceErrorDialog.prx
Needed        : libSceSysmodule.prx
Needed        : libSceUserService.prx
Needed        : libSceSystemService.prx
Needed        : libSceFiber.prx
Needed        : libSceFios2.prx
Needed        : libScePad.prx
Needed        : libSceVideoOut.prx
Needed        : libSceGnmDriver.prx
Needed        : libkernel.prx
Needed        : libSceMouse.prx
Needed        : libSceIme.prx
Needed        : libSceRtc.prx
Needed        : libSceFont-module.prx
Needed        : libSceFontFt-module.prx
Needed        : libSceAudioOut.prx
Needed        : libSceSaveData.prx
Needed        : libSceSaveDataDialog.prx
Needed        : libSceCommonDialog.prx
Needed        : libSceScreenShot.prx
Needed        : libSceGameLiveStreaming.prx
Needed        : libSceVideoRecording.prx
Needed        : libSceVideodec.prx
Needed        : libSceNet.prx
Needed        : libSceNetCtl.prx
Needed        : libSceNpCommon.prx
Needed        : libSceNpManager.prx
Needed        : libSceNpScoreRanking.prx
Needed        : libSceNpTrophy.prx
Needed        : libSceNpTus.prx
Needed        : libSceNpMatching2.prx
Needed        : libSceSsl.prx
Needed        : libSceHttp.prx
Needed        : libSceNpWebApi.prx
Needed        : libSceAppContent.prx
Needed        : libScePlayGo.prx
Needed        : libScePlayGoDialog.prx
Needed        : libSceImeBackend.prx
Needed        : libSceSharePlay.prx
Needed        : libSceJson2.prx
Needed        : libSceSigninDialog.prx
Needed        : libSceMsgDialog.prx
Needed        : libSceHmd.prx
Needed        : libSceCamera.prx
Needed        : libSceVrTracker.prx
Needed        : libSceHmdSetupDialog.prx
Needed        : libSceAjm.prx
Needed        : libSceNpCommerce.prx
Needed        : libc.prx
Module flags  : MAIN_PROG | TEXT_REL | UNK2
TLS index     : 1
Memory address: 0x000001ff8b020000:0x000001ff8c064000
Initialization: 0x000001ff8b420020
Entry address : 0x000001ff8b640730
Finalization  : 0x000001ff8b82ab00
TLS init      : 0x000001ff8b420000:0x000001ff8b420000
TLS size      : 0
EH header     : 0x000001ff8b98df54:0x000001ff8b9b1a18
EH frame      : 0x000001ff8b90d3d0:0x000001ff8b98df54
Process param : 0x000001ff8b9f8000:0x000001ff8b9f8050
Program 0 is mapped to 0x000001ff8b420000:0x000001ff8b9b4000 with EXECUTE | READ.
Program 1 is mapped to 0x000001ff8b9b4000:0x000001ff8b9f8000 with READ.
Program 2 is mapped to 0x000001ff8b9f8000:0x000001ff8be64000 with WRITE | READ.
++++++++++++++++++ I [00:00:00:00:021]:0x0000000000000258:kernel\src/main.rs:158
Loading /system/common/lib/libkernel.sprx.
++++++++++++++++++ I [00:00:00:00:022]:0x0000000000000258:kernel\src/main.rs:169
Image format  : ELF
Image type    : ET_SCE_DYNAMIC
Program 0     : 0x0000000000004000:0x0000000000054318 => 0x0000000000000000:0x0000000000050318 PT_LOAD
Program 1     : 0x0000000000058000:0x0000000000059010 => 0x0000000000054000:0x0000000000055010 PT_SCE_RELRO
Program 2     : 0x000000000005c000:0x000000000005c330 => 0x0000000000058000:0x000000000008f590 PT_LOAD
Program 3     : 0x000000000005c000:0x000000000005c020 => 0x0000000000058000:0x0000000000058020 PT_SCE_MODULE_PARAM
Program 4     : 0x000000000006d488:0x000000000006d9a8 => 0x0000000000000000:0x0000000000000520 PT_DYNAMIC
Program 5     : 0x0000000000000000:0x0000000000000000 => 0x0000000000000000:0x0000000000000000 PT_TLS
Program 6     : 0x0000000000050e9c:0x0000000000054318 => 0x000000000004ce9c:0x0000000000050318 PT_GNU_EH_FRAME
Program 7     : 0x000000000005c330:0x000000000006d9a8 => 0x0000000000000000:0x0000000000000000 PT_SCE_DYNLIBDATA
Program 8     : 0x0000000000000000:0x0000000000000000 => 0x0000000000000000:0x0000000000000000 PT_SCE_VERSION
Needed        : libSceLibcInternal.prx
Needed        : libSceGnmDriver.prx
Module flags  : UNK1
TLS index     : 0
Memory address: 0x000001ff8c070000:0x000001ff8c300000
Entry address : 0x000001ff8c0987e8
Finalization  : 0x000001ff8c0a67a0
TLS init      : 0x000001ff8c070000:0x000001ff8c070000
TLS size      : 0
EH header     : 0x000001ff8c0bce9c:0x000001ff8c0c0318
EH frame      : 0x000001ff8c0ac270:0x000001ff8c0bce9c
Program 0 is mapped to 0x000001ff8c070000:0x000001ff8c0c4000 with EXECUTE | READ.
Program 1 is mapped to 0x000001ff8c0c4000:0x000001ff8c0c8000 with READ.
Program 2 is mapped to 0x000001ff8c0c8000:0x000001ff8c100000 with WRITE | READ.
++++++++++++++++++ I [00:00:00:00:024]:0x0000000000000258:kernel\src/main.rs:180
Loading /system/common/lib/libSceLibcInternal.sprx.
++++++++++++++++++ I [00:00:00:00:026]:0x0000000000000258:kernel\src/main.rs:191
Image format  : ELF
Image type    : ET_SCE_DYNAMIC
Program 0     : 0x0000000000004000:0x0000000000129c8c => 0x0000000000000000:0x0000000000125c8c PT_LOAD
Program 1     : 0x000000000012c000:0x0000000000130510 => 0x0000000000128000:0x000000000012c510 PT_SCE_RELRO
Program 2     : 0x0000000000134000:0x000000000013dbe8 => 0x0000000000130000:0x0000000000149110 PT_LOAD
Program 3     : 0x0000000000134000:0x000000000013a4d8 => 0x0000000000130000:0x00000000001364d8 PT_SCE_MODULE_PARAM
Program 4     : 0x000000000016cc78:0x000000000016cf18 => 0x0000000000000000:0x00000000000002a0 PT_DYNAMIC
Program 5     : 0x000000000013a4e0:0x000000000013a660 => 0x00000000001364e0:0x0000000000136a40 PT_TLS
Program 6     : 0x0000000000122a88:0x0000000000129c8c => 0x000000000011ea88:0x0000000000125c8c PT_GNU_EH_FRAME
Program 7     : 0x000000000013dc00:0x000000000016cf18 => 0x0000000000000000:0x0000000000000000 PT_SCE_DYNLIBDATA
Program 8     : 0x0000000000000000:0x0000000000000000 => 0x0000000000000000:0x0000000000000000 PT_SCE_VERSION
Needed        : libkernel.prx
Needed        : libSceLibcInternal.prx
Module flags  : UNK1
TLS index     : 2
Memory address: 0x000001ff8c300000:0x000001ff8c64c000
Finalization  : 0x000001ff8c3d53b0
TLS init      : 0x000001ff8c4364e0:0x000001ff8c436660
TLS size      : 1376
EH header     : 0x000001ff8c41ea88:0x000001ff8c425c8c
EH frame      : 0x000001ff8c3f9664:0x000001ff8c41ea88
Program 0 is mapped to 0x000001ff8c300000:0x000001ff8c428000 with EXECUTE | READ.
Program 1 is mapped to 0x000001ff8c428000:0x000001ff8c430000 with READ.
Program 2 is mapped to 0x000001ff8c430000:0x000001ff8c44c000 with WRITE | READ.
++++++++++++++++++ I [00:00:00:00:028]:0x0000000000000258:kernel\src/main.rs:194
Initializing system call routines.
++++++++++++++++++ I [00:00:00:00:028]:0x0000000000000258:kernel\src/main.rs:200
Initializing execution engine.
++++++++++++++++++ I [00:00:00:00:028]:0x0000000000000258:kernel\src/main.rs:215
Patching modules.
++++++++++++++++++ I [00:00:00:00:399]:0x0000000000000258:kernel\src/main.rs:219
353 patch(es) have been applied to /system/common/lib/libkernel.sprx.
187 patch(es) have been applied to /system/common/lib/libSceLibcInternal.sprx.
2 module(s) have been patched successfully.
++++++++++++++++++ I [00:00:00:00:399]:0x0000000000000258:kernel\src/main.rs:266
Allocating application stack.
++++++++++++++++++ I [00:00:00:00:400]:0x0000000000000258:kernel\src/main.rs:284
Starting application.
</code>
</details>

```
1FF8C0999B6: 48 89 8C C4 E0 00 00 00    mov    qword ptr [rsp + 8*rax + 0xe0], rcx

Exception has occurred: Exception
Exception 0xc0000005 encountered at address 0x1ff8c0999b6: Unknown access violation (code 3) at location 0x00000000
```

**Comment by VocalFan** [comment]
image
And yes, I have the latest commit.

**Comment by ultimaweapon** [comment]
@VocalFan the latest commit should fix your issue.

**Comment by VocalFan** [comment]
All games now reach and log syscall 608 properly. It doesn't Kernel Crash before the logging is complete. So, seems good!

File: src/llt/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+
+    #[cfg(unix)]
+    #[error("cannot set thread stack")]
+    SetStackFailed(#[source] std::io::Error),
+
+    #[error("cannot create a thread")]
+    CreateThreadFailed(#[source] std::io::Error),
+}
+
+#[cfg(test)]
```
Comment (bayedieng) [inline src/llt/src/lib.rs:185]: If weare going to be adding tests maybe have it checked at CI as well?

Comment (ultimaweapon) [reply src/llt/src/lib.rs:185]: I was lazy to setup CTest but forgot that we can just run only Rust tests.

**Comment by VocalFan** [comment]
Hopped on just to see this lol

**Review by bayedieng — APPROVED** [review]

**Pull request merged**